sources/zarr/large_image_source_zarr/__init__.py

```diff
@@ -79,7 +90,47 @@ def __init__(self, path, **kwargs):
         except Exception:
             msg = 'File cannot be opened -- not an OME NGFF file or understandable zarr file.'
             raise TileSourceError(msg)
-        self._tileLock = threading.RLock()
+
+    def _initNew(self, path, **kwargs):
+        """
+        Initialize the tile class for creating a new image.
+        """
+        self._tempfile = tempfile.NamedTemporaryFile(suffix=path)
+        self._zarr_store = zarr.SQLiteStore(self._tempfile.name)
```

Comment:
I switched to using `DirectoryStore` in [5118fbc](https://github.com/girder/large_image/pull/1446/commits/5118fbc7121f470065f846a66b7ad49858c292a5). This will need further testing and refinement, though.